### PR TITLE
Components: Add `useBreakpointMatch` hook

### PR DESCRIPTION
**Pull request by renatoagds**
Insert a new hook, that will help us to identify greater, less, or current breakpoint.

This is useful for cases that we want to hide or do specifics UI changes based on breakpoint.

#### Changes proposed in this Pull Request:

* Add `useBreakpointMatch` hook.
* Add `breakpoints.module.scss` which will be our main source for breakpoints values.

#### Jetpack product discussion

1200295095943283-as-1202214413756002/f

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Start `Storybook`
* Navigate to `JS Packages/Components/Layout/Breakpoint Match`
* Start to play with window size and check if the infos are matching.

#### Demo

https://user-images.githubusercontent.com/1663717/167041217-6fbecc8f-a7e0-417c-84a7-7e58944fb8a9.mp4

### CODE REVIEW

File: projects/js-packages/components/components/layout/breakpoints.module.scss [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+@value sm: (max-width: sm-max);
+
+// MD Values
+@value md-min: 600px;
+@value md-max: 959px;
+@value md: (min-width: md-min) and (max-width: md-max);
+
+// LG Values
+@value lg-min: 960px;
+@value lg: (min-width: lg-min);
```
Comment (retrofox) [inline projects/js-packages/components/components/layout/breakpoints.module.scss:13]: I guess these values have been defined based on designs, correct?

**Review by retrofox — COMMENTED** [review]
It looks and works very well. Nice work!
I have a minor suggestion/requirement. The implementation doesn't have any document even jsdoc. A Readme.md file would be nice.

File: projects/js-packages/components/components/layout/breakpoints.module.scss (hunk 1 above)
Comment (renatoagds) [reply projects/js-packages/components/components/layout/breakpoints.module.scss:13]: Ref: PbJqbW-7N-p2

**Comment by renatoagds** [comment]
@retrofox Done in [c5b17ca](https://github.com/Automattic/jetpack/pull/24263/commits/c5b17caa77c7efb9364e72a4216059a1e371b5bd)

**Review by retrofox — APPROVED** [review]
Tested, and it works as expected. 💯

**Review by kraftbj — APPROVED** [review]

**Comment by kraftbj** [comment]
r245380-wpcom

**Pull request merged**